lua/telescope/actions/init.lua

```diff
@@ -250,6 +250,22 @@ actions.git_add = function(prompt_bufnr)
   os.execute('git add ' .. val)
 end
 
+actions.git_staging_toggle = function(prompt_bufnr)
+  local selection = actions.get_selected_entry(prompt_bufnr)
+
+  -- If parts of the file are staged and unstaged at the same time, stage
+  -- changes. Else toggle between staged and unstaged if the file is tracked,
+  -- and between added and untracked if the file is untracked.
+  if selection.status:sub(2) == ' ' then
+    os.execute('git restore --staged ' .. selection.value)
+  else
+    os.execute('git add ' .. selection.value)
+  end
+  actions.close(prompt_bufnr)
+  vim.cmd [[lua require('telescope.builtin').git_status()]]
```

Comment:
Calling vimL out of lua just to call lua out of vimL is so next level inception.
Its just `require('telescope.buitlin').git_status()`.

Also can you remove actions.git_add()? Its not used anymore